WgWall/ClientApp/src/app/tasks/tasks.component.ts

```diff
@@ -1,77 +1,168 @@
 import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
 
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import {
-    faCheck, faEyeSlash, faPencilAlt, faPlus, faSave, faTimes, faTrash, faEye
+  faCheck, faPencilAlt, faPlus, faSave, faUndo, faTrash
 } from '@fortawesome/free-solid-svg-icons';
 
 import { TaskTemplate } from '../models/task-template';
 import { TaskTemplateService } from '../services/task-template.service';
 
 @Component({
   selector: 'app-tasks',
-  templateUrl: './tasks.component.html'
+  templateUrl: './tasks.component.html',
+  styleUrls: ['./tasks.component.css']
 })
 export class TasksComponent implements OnInit {
   //icons
   public faCheck = faCheck;
   public faTrash = faTrash;
   public faPencilAlt = faPencilAlt;
   public faSave = faSave;
-  public faEyeSlash = faEyeSlash;
-  public faEye = faEye;
-  public faTimes = faTimes;
+  public faUndo = faUndo;
   public faPlus = faPlus;
 
   //task lists
-  private taskTemplates$ : Observable<TaskTemplate[]>;
-  public activeTaskTemplates$ : Observable<TaskTemplate[]>;
+  public taskTemplates: TaskTemplate[] = [];
 
-  //input
-  public editTaskTemplate: TaskTemplate = new TaskTemplate();
-  public isEditActive: boolean = false;
+  //to disable buttons when appropiate
+  public actionsActive: number;
+
+  //edit entries
+  public editSource: TaskTemplate;
+  public editContainer: TaskTemplate;
+
+  //confirm entry
+  public executeSource: TaskTemplate;
 
   constructor(private taskTemplateService: TaskTemplateService) { }
 
   ngOnInit() {
-    this.taskTemplates$ = this.taskTemplateService.get();
-    this.activeTaskTemplates$ = this.taskTemplates$.pipe(
-      map(taskTemplate => taskTemplate.filter(tt => !tt.hidden))
-    );
+    this.taskTemplateService.get().subscribe(taskTemplates => {
+      taskTemplates.forEach(tt => {
+        tt.expectedRelativeCompletion = this.calculateExpectedRelativeCompletion(tt);
+      });
+      this.taskTemplates = taskTemplates.sort((a, b) => b.expectedRelativeCompletion - a.expectedRelativeCompletion);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move this array "sort" operation to a separate statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4043)